src/pyodide/internal/_workers.py

```diff
@@ -157,13 +159,35 @@ async def blob(self) -> "Blob":
         return Blob(await self.js_object.blob())
 
 
+from js import Object
+
+
+async def _pyfetch(url: str, **kwargs: Any) -> FetchResponse:
+    # TODO: This is copied from https://github.com/pyodide/pyodide/blob/d3f99e1d/src/py/pyodide/http.py
+
+    if pyodide_version == "0.26.0a2":
+        custom_fetch = kwargs["fetcher"] if "fetcher" in kwargs else js.fetch
+        kwargs["fetcher"] = None
+        try:
+            return FetchResponse(
+                url,
+                await custom_fetch(
+                    url, to_js(kwargs, dict_converter=Object.fromEntries)
+                ),
+            )
+        except JsException as e:
+            raise OSError(e.message) from None
+    else:
+        return pyfetch(url, **kwargs)
+
+
 async def fetch(
```

Comment:
Could we rename this to something other than `fetch`? Since it's quite overloaded? Or this is intended specifically to be a public function?